drools-ansible-rulebook-integration-api/src/main/java/org/drools/ansible/rulebook/integration/api/domain/RuleGenerationContext.java

```diff
@@ -224,23 +224,23 @@ List<Rule> toExecModelRules(RulesSet rulesSet, org.drools.ansible.rulebook.integ
 	    if (getRuleName() == null) {
 	        setRuleName("r_" + ruleCounter.getAndIncrement());
 	    }
-        setRuleSetName(rulesSet.getName());
-	
+
 	    List<org.drools.model.Rule> rules = createRules(rulesExecutionController);
 	    if (hasTemporalConstraint(ansibleRule)) {
 	        rulesSet.withOptions(RuleConfigurationOption.EVENTS_PROCESSING);
 	    }
 	    return rules;
 	}
 
-	private void updateContextFromRule(org.drools.ansible.rulebook.integration.api.domain.Rule anisbleRule) {
-		setRuleName(anisbleRule.getName());
-		setAction(anisbleRule.getAction());
-		if (anisbleRule.getOptions() != null) {
-			addOptions(anisbleRule.getOptions().getOptions());
-		}
-		setCondition(anisbleRule.getCondition());
-	}
+    private void updateContextFromRule(org.drools.ansible.rulebook.integration.api.domain.Rule anisbleRule) {
+        setRuleSetName(anisbleRule.getRuleSetName());
```

Comment:
`ruleSetName` needs to be populated here. Previously, it was set in `toExecModelRules`, but runtime engine (`AstRulesEngine`, `AsyncAstRulesEngine` = used by drools_jpy) may call `rulesSet.hasTemporalConstraint` and `rulesSet.hasAsyncExecution` earlier and they create operator instances, so `ruleSetName` was null in an error log.